src/bin/common/solving_loop.ml

```diff
@@ -137,9 +137,15 @@ let main () =
         end;
       SAT.reset_refs ();
       let ftdn_env = FE.init_env used_context in
+      let hook_on_status status i =
+        Frontend.print_status status i;
+        match status with
+        | Timeout _ -> exit_as_timeout ()
```

Comment:
```suggestion
        | Timeout _ when not Options.get_timelimit_per_goal () -> exit_as_timeout ()
```

Reply:
What do you mean? There is only one timeout -- the `--timelimit-per-goal` option changes the meaning of that timeout from "per solver instance" to "per SAT query" (in which case we need to continue processing).

Reply:
Can't blame you, there should be IMO (see also the discussion in #946), but yeah currently we only have the one.